develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown

```diff
@@ -0,0 +1,221 @@
+
+                writer.print(formHtml);
+
+            }
+
+            @Override
+            public Map<String, Object> getRequestParameters(
+                HttpServletRequest request) {
+
+                Map<String, Object> params = new HashMap<>();
+
+                String[] mysymbols = ParamUtil.getParameterValues(request, "mysymbols");
+
+                params.put("symbols", mysymbols);
+
+                return params;
+            }
+        }
+
+The above example ...
+
+## Creating a Completely Custom Configuration UI
+
+In some cases, you want a completely custom UI for your configuration. For
+example: 
+
+- Your application doesn't use Config Admin to provide its configuration. You
+    have a completely different configuration backend, and you'll write a
+    completely independent frontend.
+
+- Your application needs more flexibility in its UI, such as multiple
+    configuration screens.
+
+To accomplish this, write a `ConfigurationScreen` implementation.
+
+At a high level you must
+
+1.  Write a Component that declares itself an implementation of the
+    `ConfigurationScreen` interface.
+
+2.  Implement `Configurationscreen`'s methods.
+
+Here's an example implementation:
+
+
+    @Component(immediate = true, service = ConfigurationScreen.class) 
+    public class SampleConfigurationScreen implements ConfigurationScreen {
+
+First declare the class an implementation of `ConfigurationScreen`.
+
+    @Override 
+    public String getCategoryKey() { 
+
+        return "third-party"; 
+        
+    }
+
+    @Override 
+    public String getKey() { 
+    
+        return "sample-configuration-screen"; 
+        
+    }
+
+    @Override 
+    public String getName(Locale locale) { 
+    
+        return "Sample Configuration Screen"; 
+
+    }
+
+Second, set the category key, then configuration entry's key and localized name.
+This example puts the configuration entry, keyed `sample-cofniguration-screen`,
+into the `third-party` System Settings section. The String that appears in
+System Settings is _Sample Configuration Screen_.
+
+    @Override 
+    public String getScope() { 
+    
+        return "system"; 
+        
+    }
+
+Third, set the 
+[configuration scope](/develop/tutorials/-/knowledge_base/7-1/scoping-configurations).
+
+
+    @Override 
+    public void render(HttpServletRequest request, HttpServletResponse response) 
+        throws IOException {
+
+        _jspRenderer.renderJSP( _servletContext, request, response,
+        "/sample_configuration_screen.jsp"); 
+        
+    }
+
+    @Reference private JSPRenderer _jspRenderer;
+
+    @Reference(
+        target ="(osgi.web.symbolicname=com.liferay.currency.converter.web)", 
+        unbind = "-")
+    private ServletContext _servletContext;
+
+The most important step is to write the `render` method.This example relies on
+the `JSPRenderer` service to delegate rendering to a JSP.
+
+<!--Should we provide some sample JSP markup?-->
```

Comment:
@jorgeferrer do you think some example JSP (perhaps creating the multi-page configuration form mentioned in the intro) would be helpful here?

Reply:
Yes, definitely.

I think Drew should be able to help with that.

Reply:
cc @drewbrokke, care to provide some sample JSP for the docs? Seeing a concrete example will help devs consider the possibilities of the feature.

Reply:
Hey @rbohl sorry about the delay.  We don't have any examples ready to go so I will try to get them to you during this sprint.

Reply:
@drewbrokke I have a memory of you giving me some example code recently, but I can't find it in any of the usual places. Was I mis-remembering? If so, _bump_.